src/backend/distributed/utils/maintenanced.c

```diff
@@ -270,8 +276,15 @@ CitusMaintenanceDaemonMain(Datum main_arg)
 			/*
 			 * We don't want to run the deadlock checks if there exists
 			 * any version mistmatch.
+			 *
+			 * We also want to make sure that the extension is available by checking
+			 * the existence of the required metadata tables and AccessShareLocks are
+			 * acquired on them.
```

Comment:
maybe "and AccessShareLocks are acquired on them" -> "and whether they can be locked for reads. Deadlock detection is skipped if the tables are locked (e.g. when the extension is being dropped) or no longer exist."